src/mapping/VolumeCellInterpolation.cpp

```diff
@@ -0,0 +1,114 @@
+#include "VolumeCellInterpolation.hpp"
+
+#include <Eigen/Core>
+#include <algorithm>
+#include <deque>
+#include <memory>
+#include <ostream>
+#include <unordered_set>
+#include <utility>
+
+#include "logging/LogMacros.hpp"
+#include "mapping/Mapping.hpp"
+#include "mapping/Polation.hpp"
+#include "math/differences.hpp"
+#include "mesh/Data.hpp"
+#include "mesh/Mesh.hpp"
+#include "mesh/SharedPointer.hpp"
+#include "mesh/Vertex.hpp"
+#include "query/Index.hpp"
+#include "utils/Event.hpp"
+#include "utils/Statistics.hpp"
+#include "utils/assertion.hpp"
+
+namespace precice {
+extern bool syncMode;
+
+namespace mapping {
+
+VolumeCellInterpolation::VolumeCellInterpolation(
+    Constraint constraint,
+    int        dimensions)
+    : BarycentricBaseMapping(constraint, dimensions)
+{
+  if (constraint == CONSISTENT) {
+    setInputRequirement(Mapping::MeshRequirement::FULL);
+    setOutputRequirement(Mapping::MeshRequirement::VERTEX);
+  } else if (constraint == CONSERVATIVE) {
+    setInputRequirement(Mapping::MeshRequirement::VERTEX);
+    setOutputRequirement(Mapping::MeshRequirement::FULL);
+  } else {
+    PRECICE_ASSERT(constraint == SCALEDCONSISTENT, constraint);
+    setInputRequirement(Mapping::MeshRequirement::FULL);
+    setOutputRequirement(Mapping::MeshRequirement::FULL);
```

Comment:
I guess in the end it would be nice to move this whole specification into the `Mapping` class, i.e., the base class as I read something like this all over the place in our preCICE mappings. Maybe an override for a specific `VolumeCellInterpolation` would be possible then...